python_modules/dagster-graphql/dagster_graphql/schema/schedules/schedules.py

```diff
@@ -37,13 +38,18 @@ class GrapheneSchedule(graphene.ObjectType):
     class Meta:
         name = "Schedule"
 
-    def __init__(self, graphene_info, external_schedule):
+    def __init__(self, graphene_info, external_schedule, schedule_state=None, fetched_state=False):
         self._external_schedule = check.inst_param(
             external_schedule, "external_schedule", ExternalSchedule
         )
```

Comment:
can I suggest a static factory method or something rather than this fetched_state arg?

Like GrapheneSchedule.create_from_single_schedule(external_schedule): which loads the state and passes it in to the constructor?

I like that layering a little better - so you can add schedule_state and count on it always have being fetched in some way before coming into the constructor